### ISSUE TITLE
ACDD-1.3 compliance check does not account for all_leap calendar

### ISSUE THREAD

**Issue by huaracheguarache** [ioos/compliance-checker#1249]
**Version of compliance checker running:** 5.4.2

**Describe the checker this affects:** ACDD-1.3

**Attach a minimal CDL or NetCDF file which is able to reproduce the issue**

I'm unable to attach netCDF-files to this report as .nc files are not a supported file type.

**To Reproduce:**

This can be reproduced fairly easily by running the following python code:

```python
import xarray as xr

ds = xr.tutorial.load_dataset('air_temperature')
ds.attrs['time_coverage_start'] = str(ds.time[0].dt
                                      .strftime('%Y-%m-%dT%H:%M:%SZ').values)
ds.attrs['time_coverage_end'] = str(ds.time[-1].dt
                                    .strftime('%Y-%m-%dT%H:%M:%SZ').values)
ds.attrs['Conventions'] = 'COARDS, ACDD-1.3'

ds.to_netcdf('regular_calendar.nc')

ds = ds.convert_calendar('all_leap')
units = 'hours since 1800-01-01'

ds.to_netcdf('all_leap_calendar.nc', encoding={'time': {'units': units}})

```

And then running the compliance checker on the resulting files named `regular_calendar.nc`, and `all_leap_calendar.nc`.

**Describe the issue below:**

The ACDD-1.3 compliance checker reports that the start and end of time_coverage_extent do not match the actual time coverage extent of the data when using an `all_leap` calendar. This does not happen when the calendar is `standard`.

The following is the output when running the compliance checker on `regular_calendar.nc`:

```bash
Running Compliance Checker on the datasets from: ['/home/michaely/src/jobb/IOOS_bug/regular_calendar.nc']


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                 Version 5.4.2                                  
                     Report generated 2025-10-09T09:33:58Z                      
                                    acdd:1.3                                    
http://wiki.esipfed.org/index.php?title=Category:Attribute_Conventions_Dataset_Discovery
--------------------------------------------------------------------------------
                               Corrective Actions                               
regular_calendar.nc has 6 potential issues


                               Highly Recommended                               
--------------------------------------------------------------------------------
Global Attributes
* keywords not present
* summary not present

variable "air" missing the following attributes:
* coverage_content_type
* standard_name


                                  Recommended                                   
--------------------------------------------------------------------------------
Global Attributes
* acknowledgment/acknowledgement not present
* comment not present
* creator_name not present
* creator_url not present
* creator_email not present
* geospatial_bounds not present
* geospatial_lat_min not present
* geospatial_lat_max not present
* geospatial_lon_min not present
* geospatial_lon_max not present
* geospatial_vertical_min not present
* geospatial_vertical_max not present
* geospatial_vertical_positive not present
* geospatial_bounds_crs not present
* geospatial_bounds_vertical_crs not present
* history not present
* id not present
* institution not present
* license not present
* naming_authority not present
* project not present
* processing_level not present
* publisher_name not present
* publisher_url not present
* publisher_email not present
* standard_name_vocabulary not present
* source not present
* time_coverage_duration not present
* time_coverage_resolution not present

date_created_is_iso
* Attr date_created is not present

geospatial_lat_extents_match
* geospatial_lat_min/max attribute not found, CF-1.6 spec chapter 4.1

geospatial_lon_extents_match
* geospatial_lon_min/max attribute not found, CF-1.6 spec chapter 4.1
```

And this is the output when running it on `all_leap_calendar.nc`:

```bash
Running Compliance Checker on the datasets from: ['/home/michaely/src/jobb/IOOS_bug/all_leap_calendar.nc']


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                 Version 5.4.2                                  
                     Report generated 2025-10-09T09:38:28Z                      
                                    acdd:1.3                                    
http://wiki.esipfed.org/index.php?title=Category:Attribute_Conventions_Dataset_Discovery
--------------------------------------------------------------------------------
                               Corrective Actions                               
all_leap_calendar.nc has 7 potential issues


                               Highly Recommended                               
--------------------------------------------------------------------------------
Global Attributes
* keywords not present
* summary not present

variable "air" missing the following attributes:
* coverage_content_type
* standard_name


                                  Recommended                                   
--------------------------------------------------------------------------------
Global Attributes
* acknowledgment/acknowledgement not present
* comment not present
* creator_name not present
* creator_url not present
* creator_email not present
* geospatial_bounds not present
* geospatial_lat_min not present
* geospatial_lat_max not present
* geospatial_lon_min not present
* geospatial_lon_max not present
* geospatial_vertical_min not present
* geospatial_vertical_max not present
* geospatial_vertical_positive not present
* geospatial_bounds_crs not present
* geospatial_bounds_vertical_crs not present
* history not present
* id not present
* institution not present
* license not present
* naming_authority not present
* project not present
* processing_level not present
* publisher_name not present
* publisher_url not present
* publisher_email not present
* standard_name_vocabulary not present
* source not present
* time_coverage_duration not present
* time_coverage_resolution not present

date_created_is_iso
* Attr date_created is not present

geospatial_lat_extents_match
* geospatial_lat_min/max attribute not found, CF-1.6 spec chapter 4.1

geospatial_lon_extents_match
* geospatial_lon_min/max attribute not found, CF-1.6 spec chapter 4.1

time_coverage_extents_match
* Date time mismatch between time_coverage_start and actual time values 2013-01-01T00:00:00+00:00 (time_coverage_start) != 2013-06-11T00:00:00+00:00 (time[0])
* Date time mismatch between time_coverage_end and actual time values 2014-12-31T18:00:00+00:00 (time_coverage_end) != 2015-06-12T18:00:00+00:00 (time[N])
```

Notice the time coverage extent mismatch warning at the bottom.
